plugins/statuslines/lualine.nix

```diff
@@ -1,260 +1,281 @@
+
+              icon = lib.nixvim.mkNullOrOption' {
+                type = either str (attrsOf anything);
+                description = "Defines the icon to be displayed in front of the component.";
+              };
+
+              separator = mkSeparatorsOption { };
+
+              color = defaultNullOpts.mkNullable (either attrs str) null ''
+                Defines a custom color for the component.
+
+                Default is the color defined by your theme for that section and mode.
+              '';
+
+              padding = defaultNullOpts.mkNullable (either int (submodule {
+                # In case they add support for top/bottom padding
                 freeformType = attrsOf anything;
 
-                options = {
-                  icon = mkOption {
-                    type = str;
-                    description = "Icon character.";
-                  };
-                };
-              })
-            ) "Defines the icon to be displayed in front of the component.";
-
-            separator = mkSeparatorsOption { };
-
-            color = helpers.mkNullOrOption (types.attrsOf types.str) ''
-              Defines a custom color for the component.
-            '';
-
-            padding = helpers.defaultNullOpts.mkNullable (types.either types.int (
-              types.submodule {
                 options = {
                   left = mkOption {
-                    type = types.int;
-                    description = "left padding";
+                    type = int;
+                    description = "Left padding.";
                   };
                   right = mkOption {
-                    type = types.int;
-                    description = "left padding";
+                    type = int;
+                    description = "Right padding.";
                   };
                 };
-              }
-            )) 1 "Adds padding to the left and right of components.";
-
-            fmt = helpers.mkNullOrLuaFn ''
-              A lua function to format the component string.
-
-              Example:
-              ```lua
-                function(text)
-                  return text .. "!!!"
-                end
-              ```
-            '';
-
-            extraConfig = mkOption {
-              type = types.attrs;
-              default = { };
-              description = "extra options for the component";
+              })) 1 "Adds padding to the left and right of components.";
+
+              fmt = lib.nixvim.mkNullOrLuaFn' {
+                description = ''
+                  A lua function to format the component string.
+                '';
+                example = ''
+                  ```lua
+                    function(text)
+                      return text .. "!!!"
+                    end
+                  ```
+                '';
+              };
             };
           };
-        })
-      )
-    ) "";
+        in
+        listOf (lib.nixvim.transitionType oldType coerceFn newType);
```

Comment:
Note: this won't have rawLua support. IDK if having that is worth the required complexity though...

```nix
  maybeRaw (listOf ((lib.nixvim.transitionType oldType coerceFn (maybeRaw newType))));
```

Technically, we can get away without `maybeRaw newType` because `{ __raw = ""; }` will match both `rawLua.check` and `(attrsOf anything).check`.

Having the list itself be `maybeRaw` could also be achieved using `mkNullableWithRaw` or `mkListOf`, however I suspect we will run into #1618 if we haven't already...